core-java/src/main/java/eu/neverblink/jelly/core/ProtoEncoder.java

```diff
@@ -0,0 +1,134 @@
+package eu.neverblink.jelly.core;
+
+import eu.neverblink.jelly.core.internal.ProtoEncoderBase;
+import eu.neverblink.jelly.core.proto.v1.RdfStreamOptions;
+import eu.neverblink.jelly.core.proto.v1.RdfStreamRow;
+import java.util.Collection;
+
+/**
+ * Base interface for RDF stream encoders.
+ * @param <TNode> type of RDF nodes in the library
+ */
+public abstract class ProtoEncoder<TNode>
+    extends ProtoEncoderBase<TNode>
+    implements RowBufferAppender, ProtoHandler.AnyProtoHandler<TNode> {
+
+    /**
+     * Parameters passed to the Jelly encoder.
+     * <p>
+     * New fields may be added in the future, but always with a default value and in a sequential order.
+     * However, it is still recommended to use named arguments when creating this object.
```

Comment:
This is out of date, I think.

I haven't worked with Java records. Can we add new fields in a backwards-compatible manner? Or would we need explicit constructors for that?

Reply:
we need explicit constructors for that

Reply:
Then I'm for adding explicit constructors. This will be wordy and awful, I know, but otherwise introducing a new parameter to the constructor would be a breaking change (!!!!) requiring us to bump the version to 4.0.0. That's bad.

Reply:
We can use factory methods like you did in scala,
or we can use `static Params of(...)` in Params (aka "named" constructors) to achieve same effect

Reply:
Sure, you can go with that.